[Ashiqur Rahman] iP

AshiqurRah opened the pull request:
# DukeAce
>  “Action is the foundational key to all success.” – Pablo Picasso ([source](https://dansilvestre.com/productivity-quotes))

DukeAce frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/381).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
#

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

AshiqurRah committed:
Level 1

AshiqurRah committed:
Level 2

AshiqurRah committed:
Level 3

AshiqurRah committed:
Level 4

AshiqurRah committed:
A-TextUiTesting

AshiqurRah committed:
Level 5

AshiqurRah committed:
Level 6

AshiqurRah committed:
Minor edits

AshiqurRah committed:
Level 7, Save feature done in branch

AshiqurRah committed:
Branch Level 8

AshiqurRah committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level 7, Save feature done in branch

AshiqurRah committed:
Merge branch 'branch-Level-8'

AshiqurRah committed:
A-MoreOOP

AshiqurRah committed:
A-Packages

AshiqurRah committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

AshiqurRah committed:
A-JUnit

AshiqurRah committed:
A-jar

AshiqurRah committed:
Minor Edits to Storage

AshiqurRah committed:
A-JavaDoc

AshiqurRah committed:
A-CodingStandard

AshiqurRah committed:
Level-9

AshiqurRah committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc

AshiqurRah committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

# Conflicts:
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Ui.java

AshiqurRah committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

AshiqurRah committed:
Minor Edits in Storage

leongdl135 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,163 @@
+                    echo(taskStatus);
+                    writeToFile(tasklist);
+                }
+            } else if (command.contains("mark")){ // to detect mark command
```
Should there be a spacing?
```suggestion
            } else if (command.contains("mark")) { // to detect mark command
```

leongdl135 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,163 @@
+            } else if (command.contains("mark")){ // to detect mark command
+                String number = command.replaceAll("[^\\d.]", "");
+                int n = Integer.parseInt(number);
+                if (n > tasklist.size()){
```
Should there be a spacing?
```suggestion
                if (n > tasklist.size()) {
```

leongdl135 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+ * @author Ashiqur Rahman A0230107Y
+ */
+public class TaskList {
+    protected ArrayList<Task> allTasks;
```
I like how you named your ArrayList<> allTasks instead of allTask 👍

leongdl135 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,163 @@
+                        writeToFile(tasklist);
+                    }
+                }
+            } else if(command.contains("find")) {
```
Should there be a space here? I noticed the same issue in several other places too.
```suggestion
            } else if (command.contains("find")) {
```

leongdl135 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                        Deadline newTask = new Deadline(description, modifiedString);
+                        tasklist.add(newTask);
+                    } else {
+                        // replace the formats with empty string
+                        String deadlineDsc = modifiedLine.replace("[D][X] ", "");
+                        // get 2 substring for description and deadlineTime
+                        String description = deadlineDsc.substring(0, deadlineDsc.indexOf("(") - 1);
+                        String dlTime = deadlineDsc.substring(deadlineDsc.indexOf("(")+1,deadlineDsc.indexOf(")"));
+                        String modDeadlineTime = dlTime.replaceFirst("by: ", "");
+                        // to convert string to date
+                        DateTimeFormatter convertFormatter = DateTimeFormatter.ofPattern("MMM d yyyy");
+                        LocalDate deadlineDate = LocalDate.parse(modDeadlineTime, convertFormatter);
+                        // to put in the same format as Deadline class parameter
+                        DateTimeFormatter stringFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                        String modifiedString = deadlineDate.format(stringFormatter);
+                        Deadline newTask = new Deadline(description, modifiedString);
+                        newTask.markAsDone();
+                        tasklist.add(newTask);
```
I like the use of comments to illustrate your code which improves readability. 👍

leongdl135 reviewed (commented):
Overall, I found your code easy to read for the most part except for a few scenarios where there were very long if else chunks. I noted a few minor coding standard violations such as extra spaces or insufficient spaces. Apart from these minor mistakes, your code fits the coding standard and looks good as a whole. 👍

AshiqurRah committed:
Level-10

AshiqurRah committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10

AshiqurRah committed:
Minor edits

AshiqurRah committed:
Updated MainClassName field in gradle build file

AshiqurRah committed:
Unable to check for errors and their actual source of error

Need to check for errors and their actual source of error to do a quick bug fix.

Let's add assertions as it will show us the variable or function that is returning the desired input.

AshiqurRah committed:
Code Quality: Long code and deep nested

Understandability is Important. Might not be able to understand code in the future.

Let's extract out codes from method that are longer than 30 lines and change the use of deep nesting.

AshiqurRah committed:
Merge pull request #2 from AshiqurRah/branch-A-Assertions

Unable to check for errors and their actual source of error

AshiqurRah committed:
Merge branch 'master' of github.com:AshiqurRah/ip

* 'master' of github.com:AshiqurRah/ip:
  Unable to check for errors and their actual source of error

AshiqurRah committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Unable to check for errors and their actual source of error

AshiqurRah committed:
Merge pull request #3 from AshiqurRah/branch-A-CodeQuality

Branch A-CodeQuality

AshiqurRah committed:
Merge branch 'master' of github.com:AshiqurRah/ip

* 'master' of github.com:AshiqurRah/ip:
  Code Quality: Long code and deep nested

AshiqurRah committed:
C-MassOps: Included extension that will mark, unmark, delete all tasks

AshiqurRah committed:
A-CI

AshiqurRah committed:
Merge branch 'master' of github.com:AshiqurRah/ip

* 'master' of github.com:AshiqurRah/ip:
  A-CI

AshiqurRah committed:
Updated Continuous Integration file

AshiqurRah committed:
A-Personality

AshiqurRah committed:
A-MoreTesting

AshiqurRah committed:
A-BetterGUI

AshiqurRah committed:
Upload Ui.png

AshiqurRah committed:
A-UserGuide

AshiqurRah committed:
Minor edits to ReadMe.md file

AshiqurRah committed:
Updated ReadMe.md

AshiqurRah committed:
Update README.md

AshiqurRah committed:
Final Updates to iP

AshiqurRah committed:
Update README.md

AshiqurRah committed:
Update READMe.md

Fixed the Command Summary syntax

AshiqurRah committed:
Update exit command

AshiqurRah committed:
Update Welcome message

AshiqurRah committed:
Fix the deadline bug

AshiqurRah committed:
Fix Event command bugs

AshiqurRah committed:
Fix Index bugs

AshiqurRah committed:
Fix event command bugs

AshiqurRah committed:
Update README.md

AshiqurRah committed:
Update start page

AshiqurRah committed:
Fix event bugs
